site/src/components/PortForwardButton/PortForwardButton.tsx

```diff
@@ -0,0 +1,156 @@
+import Button from "@material-ui/core/Button"
+import Link from "@material-ui/core/Link"
+import Popover from "@material-ui/core/Popover"
+import { makeStyles } from "@material-ui/core/styles"
+import TextField from "@material-ui/core/TextField"
+import OpenInNewOutlined from "@material-ui/icons/OpenInNewOutlined"
+import { ChooseOne, Cond } from "components/Conditionals/ChooseOne"
+import { Stack } from "components/Stack/Stack"
+import { useRef, useState } from "react"
+import { colors } from "theme/colors"
+import { CodeExample } from "../CodeExample/CodeExample"
+import { HelpTooltipLink, HelpTooltipLinksGroup, HelpTooltipText } from "../Tooltips/HelpTooltip"
+
+export interface PortForwardButtonProps {
+  host: string
+  username: string
+  workspaceName: string
+  agentName: string
+}
+
+const EnabledView: React.FC<PortForwardButtonProps> = (props) => {
+  const { host, workspaceName, agentName, username } = props
+  const styles = useStyles()
+  const [port, setPort] = useState("3000")
+  const { location } = window
+  const urlExample = `${location.protocol}//${port}--${agentName}--${workspaceName}--${username}.${host}`
+
+  return (
+    <Stack direction="column" spacing={1}>
+      <HelpTooltipText>
+        You can port forward this resource by typing the{" "}
+        <strong>port, workspace name, agent name</strong> and <strong>your username</strong> in the
+        URL like the example below
+      </HelpTooltipText>
+
+      <CodeExample code={urlExample} />
+
+      <HelpTooltipText>Or you can use the following form to open it in a new tab.</HelpTooltipText>
+
+      <Stack direction="row" spacing={1} alignItems="center">
+        <TextField
+          label="Port"
+          type="number"
+          value={port}
+          className={styles.portField}
+          onChange={(e) => {
+            setPort(e.currentTarget.value)
+          }}
+        />
+        <Link
+          underline="none"
+          href={urlExample}
+          target="_blank"
+          rel="noreferrer"
+          className={styles.openUrlButton}
+        >
+          <Button>Open URL</Button>
+        </Link>
+      </Stack>
+
+      <HelpTooltipLinksGroup>
+        <HelpTooltipLink href="https://coder.com/docs/coder-oss/latest/port-forward">
+          Learn more about port forward
+        </HelpTooltipLink>
+      </HelpTooltipLinksGroup>
+    </Stack>
+  )
+}
+
+const DisabledView: React.FC<PortForwardButtonProps> = () => {
+  return (
+    <Stack direction="column" spacing={1}>
+      <HelpTooltipText>
+        <strong>Looks like your deployment does not have port forward enabled.</strong> Check the
+        docs for more info on how to enable it.
+      </HelpTooltipText>
```

Comment:
@bpmct I would appreciate your help to improve the verbiage.

Reply:
Left a review!